noita_mod/core/files/scripts/ban_perks_events.lua

```diff
@@ -0,0 +1,246 @@
+                    local py = y - math.sin(angle) * length
+                    -- Maintain compatibility with NT_sync_perks and NT_team_perks by adding NT_DONT_SHARE
+                    local perk_entity = perk_spawn_with_name(px, py, pid, true)
+                    EntityAddComponent(perk_entity, "VariableStorageComponent",
+                        {
+                            name = "NT_DONT_SHARE",
+                            value_string = "",
+                        })
+
+                    angle = angle + inc
+                    column = column + 1
+
+                    if (column >= col_limit) then
+                        column = 0
+                        col_limit = col_limit + col_size_inc
+                        length = length + len_inc
+
+                        angle = (full_arc) / (col_limit + 1)
+                        inc = angle
+                    end
+                end
+            end
+        end
+    end
+end
+
+local function RemoveBannedPerks(player_id)
+    local perk_was_removed = false
+    -- Not sure how a player might still be nil but play it safe
+    if (player_id ~= nil) then
+        for i, perk_data in ipairs(BannedPerksDataTable()) do
+            local perk_id = perk_data.id
+
+            local flag_name = get_perk_picked_flag_name(perk_id)
+            local pickup_count = tonumber(GlobalsGetValue(flag_name .. "_PICKUP_COUNT", "0"))
+
+            if (GameHasFlagRun(flag_name) or (pickup_count > 0)) and ((perk_data.do_not_remove == nil) or (perk_data.do_not_remove == false)) then
+                perk_was_removed = true
+                print("Removing " .. perk_id)
+                GameRemoveFlagRun(flag_name)
+                pickup_count = 0
+                GlobalsSetValue(flag_name .. "_PICKUP_COUNT", tostring(pickup_count))
+
+                if (perk_data.game_effect ~= nil) then
+                    -- local comp = GameGetGameEffect( player_id, perk_data.game_effect )
+
+                    -- if ( comp ~= NULL_ENTITY ) then
+                    -- 	ComponentSetValue2( comp, "frames", 1 )
+                    -- end
+
+                    local n = GameGetGameEffectCount(player_id, perk_data.game_effect)
+                    for m = 1, n do
+                        local effect_comp = GameGetGameEffect(player_id, perk_data.game_effect)
+                        if (effect_comp ~= NULL_ENTITY) then
+                            -- Make sure to check for the perk_component so that it does not interfere with hourglass buff
+                            if ComponentHasTag(effect_comp, "perk_component") then
+                                EntityRemoveComponent(player_id, effect_comp)
+                            end
+                        end
+                    end
+                end
+
+                if (perk_data.game_effect2 ~= nil) then
+                    -- local comp = GameGetGameEffect( player_id, perk_data.game_effect2 )
```

Comment:
commented code like this should be removed, or add a comment to why it could be beneficial to keep in here